### ISSUE TITLE
Add a vite based server/middleware for our mocks

### ISSUE THREAD

**Issue by johncowen** [kumahq/kuma-gui#2001]
### Description

Everytime I add new mocks that are slightly more involved, I wish I could just navigate to the API endpoint in my browser and view the response so I can see what is happening.

Currently we can't do this because we use MSW so you have to go to a page in the GUI that you know makes the HTTP request you are building out and view the response in web inspector. If this is a new response that doesn't have any code in the GUI to request it yet, then you have to build this out first.

Vite supports serving up other files other than our HTML and JS files, and I think (from what I remember) it allows you to add middlewares for specific endpoints. We should be able to repurpose our existing mock utility functions to make a middleware for Vite also, just like we do to run them via Cypress. (also I originally had our mocks running in an express server before bringing them to MSW). This would allow us to just navigate to the URL of the API endpoint you are building and see the response.

Whilst it doesn't have to be done here, if you have this running we may aswell not use MSW during local development work and only use it for the PR previews.

---

Decided to make an issue for this rather than run off and find the express code I had and move it to Vite or just think "I'll do it next time I'm building out an endpoint" which never happens 😆

**Comment by lahabana** [comment]
@johncowen please complete with what's left

**Comment by johncowen** [comment]
This currently works fine in Kuma, and is actually running:

From https://github.com/kumahq/kuma-gui/pull/2314: 

> To try this out you need to set KUMA_API_URL=http://localhost:8080. 8080 being the port that vite is using, and also set KUMA_MOCK_API_ENABLED=false to turn off MSW.

Ideally though we need to make this work with our service injection framework, along with vite.

We don't use our service injection currently with vite anywhere. In order to use our service injection framework with vite, it cannot use any of vite's `@` import aliasing capabilities. We should where possible use "native" package names for the things we need here (`fake-api` and our service injection stuff) i.e. `@kumahq/fake-api` etc.

This then hits another problem in that it seems to be non-trivial to write a package in a workspace and share in inside the workspace without having a TS complication step. The good news is, its seems like vite6 (potentially combined with nodes new TS stripping functionality) means it will be easier to implement this.

---

Possibly blocked by https://github.com/kumahq/kuma-gui/pull/3572, the upgrade here includes new functionality to make it much easier to load workspace'd modules without compiling the Typescript first

**Comment by schogges** [comment]
I've created a follow-up issue for the next iteration and I'll close this one.

@johncowen we can potentially create an umbrella for the bigger picture?
